src/app/content/utils.ts

```diff
@@ -57,13 +58,14 @@ export function getContentPageReferences(book: ArchiveBook, page: ArchivePage) {
   return matches;
 }
 
-export const parseContents = (book: Book, contents: Array<ArchiveTree | ArchiveTreeNode>) => {
-  contents.map((subtree) => {
+export const parseContents = async(book: Book, contents: Array<ArchiveTree | ArchiveTreeNode>) => {
+  // we use 1 here to prevent the number of queued parseContents from growing exponentially
+  // if there are multiple levels in the tree
+  await asyncPool(1, contents, async(subtree: ArchiveTree | ArchiveTreeNode) => {
     subtree.title = getTitleFromArchiveNode(book, subtree);
     if (isArchiveTree(subtree)) {
-      subtree.contents = parseContents(book, subtree.contents);
```

Comment:
`parseContents` just returns the `contents` argument, so once again that works by the argument being modified in place.